### PR TITLE
New Feature OnResetStack

### PR DESCRIPTION
**Pull request by santiagorobra**
## Summary

The purpose of this pr is to add a new feature that is used many times in projects. 
Also changes were made in the routes file to improve the typing.

## New Feature
OnResetStack - it is a function that resets the stack

## New file 
routesParamList - it was necessary to follow the structure of [react navigation v5](https://reactnavigation.org/docs/typescript/)

### CODE REVIEW

File: generators/app/templates/src/constants/routesParamList.ejs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Routes from '@constants/routes';
+
+export type RoutesParamList = {
+  <%_ if(features.loginandsignup) { _%>
+  [Routes.Auth]: undefined;
+  [Routes.Login]: undefined;
+  [Routes.SignUp]: undefined;
+  <%_ } _%>
+  <%_ if(features.onboarding) { _%>
+  [Routes.OnBoarding]: undefined,
```
Comment (sfernandez11) [inline generators/app/templates/src/constants/routesParamList.ejs:10]: `[Routes.OnBoarding]: undefined;`

**Review by sfernandez11 — COMMENTED** [review]
Great work Santi! Just one comment...

File: generators/app/templates/src/constants/routesParamList.ejs (hunk 1 above)
Comment (santiagorobra) [reply generators/app/templates/src/constants/routesParamList.ejs]: thanks :D

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**